forem/forem
# Crayons DateRangePicker component - initial beta storybook component

## What type of PR is this? (check all applicable)


## Description

This PR adds an initial DateRangePicker component to Storybook under the "BETA" folder. It does not yet cover all of the functionality required in https://github.com/forem/forem/issues/17493, but I'd like to keep the PR reviews manageable, and since this only exists in the BETA storybook folder for now, I think we're good to review this initial feature set.

## Related Tickets & Documents



- Related Issue https://github.com/forem/forem/issues/17493

## QA Instructions, Screenshots, Recordings

- Locally run `yarn install && yarn storybook`
- Navigate to the BETA > DateRangePicker story

Experiment with the following features:
- Ability to set a minimum (earliest) selectable date
- Ability to set a maximum (latest) selectable date
- Ability to skip to a specific month/year from valid options via the selects
- Ability to move forward or backwards a month from the arrow buttons
- Navigation to previous/next month should become disabled when at the edge of the valid date range
- Use keyboard or mouse to select your date range


All other features noted on https://github.com/forem/forem/issues/17493 will be added in subsequent PRs.


https://user-images.githubusercontent.com/20773163/169801969-0384fbdc-55ef-4ddb-8f42-151495873282.mp4


### UI accessibility concerns?

Yes! I've tested this with Safari and VoiceOver and all seems to work as react-dates claims. In terms of testing, experiment with using the keyboard to navigate around and select your dates.

## Added/updated tests?

- [X] No, and this is why: I want to bring these in when the full feature set is ready to test and we move the component out of the BETA folder

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Storybook](https://storybook.forem.com/) (for Crayons components)
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
- [X] This change does not need to be communicated, and this is why not: just a first step towards the overall issue



## [optional] What gif best describes this PR or how it makes you feel?

![calendar](https://media.giphy.com/media/xT5LML4FY3If24Bf0I/giphy.gif)
